Source/Runtime/LLVMEmitIR.cpp

```diff
@@ -178,8 +178,12 @@ namespace LLVMJIT
 			// If the address is 32-bits, zext it to 64-bits.
 			// This is crucial for security, as LLVM will otherwise implicitly sign extend it to 64-bits in the GEP below,
 			// interpreting it as a signed offset and allowing access to memory outside the sandboxed memory range.
+#if INTPTR_MAX == INT32_MAX
+			auto byteIndex = dispatch(*this,address,TypeId::I32);
+#else
```

Comment:
This needs to call dispatch with the correct TypeId (including TypeId::I64 if isFarAddress). Then it can use CreateTrunc to discard the upper half of a 64-bit address.